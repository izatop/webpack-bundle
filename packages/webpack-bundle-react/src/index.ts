
interface IValue {
    up: number;
    down: number;
    threshold: number;
}

const map: IValue[] = [];
for (let i = 10; i < 100; i = i + 2) {
    map.push({
        down: i - 3,
        threshold: 1,
        up: i,
    });
}
