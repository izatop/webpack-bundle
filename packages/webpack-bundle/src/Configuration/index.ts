export * from "./Options/AMD";
export * from "./Options/Bail";
export * from "./Options/Cache";
export * from "./Options/Context";
export * from "./Options/DevTool";
export * from "./Options/DevServer";
export * from "./Options/Entry";
export * from "./Options/Externals";
export * from "./Options/Mode";
export * from "./Options/Module";
export * from "./Options/ModuleRule";
export * from "./Options/Name";
export * from "./Options/Node";
export * from "./Options/Optimization";
export * from "./Options/Output";
export * from "./Options/Parallelism";
export * from "./Options/Performance";
export * from "./Options/Plugins";
export * from "./Options/Profile";
export * from "./Options/RecordsInputPath";
export * from "./Options/RecordsOutputPath";
export * from "./Options/RecordsPath";
export * from "./Options/Resolve";
export * from "./Options/ResolveLoader";
export * from "./Options/Stats";
export * from "./Options/Target";
export * from "./Options/Watch";
export * from "./Options/WatchOptions";
