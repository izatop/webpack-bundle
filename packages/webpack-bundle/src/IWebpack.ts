import {Configuration} from "webpack";

export type IWebpackKey = keyof Configuration;
export type IWebpackValue<K extends IWebpackKey = IWebpackKey> = Configuration[K];
