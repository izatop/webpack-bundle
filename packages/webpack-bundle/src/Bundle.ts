import * as webpack from "webpack";
import {BundleOption} from "./Configuration/BundleOption";

export class Bundle {
    protected options: BundleOption[] = [];

    constructor(...option: BundleOption[]) {
        this.set(...option);
    }

    public set(...option: BundleOption[]) {
        this.options.push(...option);
    }

    public get<T extends BundleOption>(type: new() => T): T | undefined {
        return this.options.find((item) => item instanceof type) as T;
    }

    public getWebpackConfig() {
        const config: webpack.Configuration = {};
        for (const option of this.options) {
            Object.assign(config, option.serialize());
        }

        return config;
    }
}
