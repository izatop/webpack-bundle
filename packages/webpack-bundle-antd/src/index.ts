import {AntdBundle} from "./Bundle/AntdBundle";

export * from "./Bundle/AntdBundle";

export default AntdBundle;
