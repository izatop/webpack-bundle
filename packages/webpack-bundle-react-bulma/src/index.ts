import {ReactBulmaBundle} from "./Bundle/ReactBulmaBundle";

export * from "./Bundle/ReactBulmaBundle";
export default ReactBulmaBundle;
