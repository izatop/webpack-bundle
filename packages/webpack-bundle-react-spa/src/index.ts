import {ReactSPABundle} from "./Bundle/ReactSPABundle";

export * from "./Bundle/ReactSPABundle";
export default ReactSPABundle;
