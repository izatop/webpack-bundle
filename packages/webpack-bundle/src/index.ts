import {Bundle} from "./Bundle";
import * as Options from "./Configuration";
import {Loader} from "./Loader/Loader";

export {Bundle, Loader, Options};
export * from "./Configuration/Option";
export * from "./Configuration/ScalarOption";
export * from "./Configuration/ObjectOption";
export * from "./Configuration/BundleOption";
export * from "./IWebpack";
