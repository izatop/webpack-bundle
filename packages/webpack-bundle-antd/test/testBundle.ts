import {AntdBundle} from "../src";

it("ReactSPABundle should generate valid configuration", () => {
    const config1 = new AntdBundle(module)
        .getWebpackConfig();

    expect(config1.mode)
        .toEqual(process.env.NODE_ENV || "development");
    expect(config1)
        .toMatchSnapshot();

    const config2 = new AntdBundle(module, "production")
        .getWebpackConfig();

    expect(config2.mode)
        .toEqual("production");
    expect(config2)
        .toMatchSnapshot();
});
