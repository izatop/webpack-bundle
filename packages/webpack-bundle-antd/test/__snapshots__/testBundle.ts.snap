// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactSPABundle should generate valid configuration 1`] = `
Object {
  "context": "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/src",
  "devServer": Object {
    "compress": true,
    "contentBase": "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/dist",
    "historyApiFallback": true,
    "hot": true,
    "port": 9000,
  },
  "entry": "entry",
  "mode": "test",
  "module": Object {
    "rules": Array [
      Object {
        "loader": "ts",
        "options": Object {
          "getCustomTransformers": [Function],
          "transpileOnly": true,
        },
        "test": /\\\\\\.tsx\\?/,
      },
      Object {
        "loader": "file",
        "options": Object {
          "name": "[hash].[ext]",
          "outputPath": "assets/",
        },
        "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\|svg\\|jpg\\|jpeg\\|png\\|gif\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
      },
      Object {
        "test": /\\\\\\.less\\$/,
        "use": Array [
          "/Users/arturbier/Development/public/nobus/webpack/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
          },
          Object {
            "loader": "less-loader",
            "options": Object {
              "javascriptEnabled": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          "/Users/arturbier/Development/public/nobus/webpack/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "splitChunks": Object {
      "cacheGroups": Object {
        "vendor": Object {
          "chunks": "all",
          "name": "vendor",
          "priority": 0,
          "test": /\\\\/node_modules\\\\//,
        },
      },
    },
  },
  "output": Object {
    "chunkFilename": "chunks/[name].js?[hash:6]",
    "filename": "[name].js?[hash:6]",
    "path": "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/dist",
    "publicPath": "/",
  },
  "plugins": Array [
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "assets/[name].css",
        "filename": "assets/[name].css",
      },
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": true,
        "meta": Object {},
        "minify": false,
        "showErrors": true,
        "template": "/Users/arturbier/Development/public/nobus/webpack/node_modules/html-webpack-plugin/default_index.ejs",
        "templateParameters": [Function],
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    [Function],
  ],
  "resolve": Object {
    "extensions": Array [
      ".ts",
      ".tsx",
      ".js",
      ".jsx",
    ],
    "modules": Array [
      "node_modules",
      "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/src",
    ],
  },
  "resolveLoader": Object {
    "modules": Array [
      "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/node_modules",
      "node_modules",
    ],
  },
}
`;

exports[`ReactSPABundle should generate valid configuration 2`] = `
Object {
  "context": "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/src",
  "entry": "entry",
  "mode": "production",
  "module": Object {
    "rules": Array [
      Object {
        "loader": "ts",
        "options": Object {
          "getCustomTransformers": [Function],
          "transpileOnly": false,
        },
        "test": /\\\\\\.tsx\\?/,
      },
      Object {
        "loader": "file",
        "options": Object {
          "name": "[hash].[ext]",
          "outputPath": "assets/",
        },
        "test": /\\\\\\.\\(woff\\(2\\)\\?\\|ttf\\|eot\\|svg\\|jpg\\|jpeg\\|png\\|gif\\)\\(\\\\\\?v=\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+\\)\\?\\$/,
      },
      Object {
        "test": /\\\\\\.less\\$/,
        "use": Array [
          "/Users/arturbier/Development/public/nobus/webpack/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
          },
          Object {
            "loader": "less-loader",
            "options": Object {
              "javascriptEnabled": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          "/Users/arturbier/Development/public/nobus/webpack/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "splitChunks": Object {
      "cacheGroups": Object {
        "vendor": Object {
          "chunks": "all",
          "name": "vendor",
          "priority": 0,
          "test": /\\\\/node_modules\\\\//,
        },
      },
    },
  },
  "output": Object {
    "chunkFilename": "chunks/[name].js?[hash:6]",
    "filename": "[name].js?[hash:6]",
    "path": "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/dist",
    "publicPath": "/",
  },
  "plugins": Array [
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "assets/[name].css",
        "filename": "assets/[name].css",
      },
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "chunksSortMode": "auto",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": true,
        "meta": Object {},
        "minify": false,
        "showErrors": true,
        "template": "/Users/arturbier/Development/public/nobus/webpack/node_modules/html-webpack-plugin/default_index.ejs",
        "templateParameters": [Function],
        "title": "Webpack App",
        "xhtml": false,
      },
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".ts",
      ".tsx",
      ".js",
      ".jsx",
    ],
    "modules": Array [
      "node_modules",
      "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/test/src",
    ],
  },
  "resolveLoader": Object {
    "modules": Array [
      "/Users/arturbier/Development/public/nobus/webpack/packages/webpack-bundle-antd/node_modules",
      "node_modules",
    ],
  },
}
`;
